### PR TITLE
PLATFORM-2393: add a default value for an i18n message in Masthead::getDefaultAvatars

### PR DESCRIPTION
**Pull request by macbre**
Prevent `VignetteRequest.php:58 - missing key 'relative-path'`" exception when the messages cache is outdated.

@nandy-andy / @wladekb / @mixth-sense

### CODE REVIEW

File: extensions/wikia/UserProfilePageV3/Masthead.class.php [hunk 1]
```diff
@@ -122,7 +122,7 @@ public function getDefaultAvatars( $thumb = "" ) {
 		wfProfileIn( __METHOD__ );
 
 		$this->mDefaultAvatars = array();
-		$images = getMessageForContentAsArray( 'blog-avatar-defaults' );
+		$images = getMessageForContentAsArray( 'blog-avatar-defaults' ) ?: ['Avatar.jpg']; // PLATFORM-2393: add a default value
```
Comment (WikiaTech) [inline extensions/wikia/UserProfilePageV3/Masthead.class.php]: INFO Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 23 issues
* MAJOR 16 major
* MINOR 1 minor
* INFO 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Masthead.class.php#L3](https://github.com/Wikia/app/blob/c05af9e4930275b6faa11ac5d5e4bcaad31a8534/extensions/wikia/UserProfilePageV3/Masthead.class.php#L3): Class "Masthead" has 30 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MAJOR [Masthead.class.php#L23](https://github.com/Wikia/app/blob/c05af9e4930275b6faa11ac5d5e4bcaad31a8534/extensions/wikia/UserProfilePageV3/Masthead.class.php#L23): Remove this unused "$mMsgCityId" private field. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1068)
1. MAJOR [Masthead.class.php#L130](https://github.com/Wikia/app/blob/c05af9e4930275b6faa11ac5d5e4bcaad31a8534/extensions/wikia/UserProfilePageV3/Masthead.class.php#L130): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [Masthead.class.php#L170](https://github.com/Wikia/app/blob/c05af9e4930275b6faa11ac5d5e4bcaad31a8534/extensions/wikia/UserProfilePageV3/Masthead.class.php#L170): Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. MAJOR [Masthead.class.php#L172](https://github.com/Wikia/app/blob/c05af9e4930275b6faa11ac5d5e4bcaad31a8534/extensions/wikia/UserProfilePageV3/Masthead.class.php#L172): Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. MAJOR [Masthead.class.php#L218](https://github.com/Wikia/app/blob/c05af9e4930275b6faa11ac5d5e4bcaad31a8534/extensions/wikia/UserProfilePageV3/Masthead.class.php#L218): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [Masthead.class.php#L255](https://github.com/Wikia/app/blob/c05af9e4930275b6faa11ac5d5e4bcaad31a8534/extensions/wikia/UserProfilePageV3/Masthead.class.php#L255): Use "static" keyword instead of "self". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)
1. MAJOR [Masthead.class.php#L380](https://github.com/Wikia/app/blob/c05af9e4930275b6faa11ac5d5e4bcaad31a8534/extensions/wikia/UserProfilePageV3/Masthead.class.php#L380): Remove the unused function parameter "$tracker". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. MAJOR [Masthead.class.php#L436](https://github.com/Wikia/app/blob/c05af9e4930275b6faa11ac5d5e4bcaad31a8534/extensions/wikia/UserProfilePageV3/Masthead.class.php#L436): Replace "and" with "&&". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2010)
1. MAJOR [Masthead.class.php#L519](https://github.com/Wikia/app/blob/c05af9e4930275b6faa11ac5d5e4bcaad31a8534/extensions/wikia/UserProfilePageV3/Masthead.class.php#L519): Remove this unused private "getThumbPath" method. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1144)

**Comment by nandy-andy** [comment]
LGTM, nitpicking: consider moving `'Avatar.jpg'` to a const ;)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 20 issues
* MAJOR 13 major
* MINOR 1 minor
* INFO 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Masthead.class.php#L3](https://github.com/Wikia/app/blob/7dd54359dee6f4869e7195c5ee10b1ce566af79f/extensions/wikia/UserProfilePageV3/Masthead.class.php#L3): Class "Masthead" has 30 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MAJOR [Masthead.class.php#L23](https://github.com/Wikia/app/blob/7dd54359dee6f4869e7195c5ee10b1ce566af79f/extensions/wikia/UserProfilePageV3/Masthead.class.php#L23): Remove this unused "$mMsgCityId" private field. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1068)
1. MAJOR [Masthead.class.php#L170](https://github.com/Wikia/app/blob/7dd54359dee6f4869e7195c5ee10b1ce566af79f/extensions/wikia/UserProfilePageV3/Masthead.class.php#L170): Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. MAJOR [Masthead.class.php#L172](https://github.com/Wikia/app/blob/7dd54359dee6f4869e7195c5ee10b1ce566af79f/extensions/wikia/UserProfilePageV3/Masthead.class.php#L172): Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. MAJOR [Masthead.class.php#L380](https://github.com/Wikia/app/blob/7dd54359dee6f4869e7195c5ee10b1ce566af79f/extensions/wikia/UserProfilePageV3/Masthead.class.php#L380): Remove the unused function parameter "$tracker". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. MAJOR [Masthead.class.php#L436](https://github.com/Wikia/app/blob/7dd54359dee6f4869e7195c5ee10b1ce566af79f/extensions/wikia/UserProfilePageV3/Masthead.class.php#L436): Replace "and" with "&&". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2010)
1. MAJOR [Masthead.class.php#L519](https://github.com/Wikia/app/blob/7dd54359dee6f4869e7195c5ee10b1ce566af79f/extensions/wikia/UserProfilePageV3/Masthead.class.php#L519): Remove this unused private "getThumbPath" method. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1144)
1. MAJOR [Masthead.class.php#L592](https://github.com/Wikia/app/blob/7dd54359dee6f4869e7195c5ee10b1ce566af79f/extensions/wikia/UserProfilePageV3/Masthead.class.php#L592): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. MAJOR [Masthead.class.php#L661](https://github.com/Wikia/app/blob/7dd54359dee6f4869e7195c5ee10b1ce566af79f/extensions/wikia/UserProfilePageV3/Masthead.class.php#L661): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
1. MAJOR [Masthead.class.php#L670](https://github.com/Wikia/app/blob/7dd54359dee6f4869e7195c5ee10b1ce566af79f/extensions/wikia/UserProfilePageV3/Masthead.class.php#L670): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)

File: extensions/wikia/UserProfilePageV3/Masthead.class.php [hunk 2]
```diff
@@ -122,12 +124,12 @@ public function getDefaultAvatars( $thumb = "" ) {
 		wfProfileIn( __METHOD__ );
 
 		$this->mDefaultAvatars = array();
-		$images = getMessageForContentAsArray( 'blog-avatar-defaults' );
+		$images = getMessageForContentAsArray( 'blog-avatar-defaults' ) ?: [ static::DEFAULT_AVATAR_FILENAME ]; // PLATFORM-2393: add a default value
```
Comment (WikiaTech) [inline extensions/wikia/UserProfilePageV3/Masthead.class.php:127]: INFO Move this trailing comment on the previous empty line. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS139)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 20 issues
* MAJOR 13 major
* MINOR 1 minor
* INFO 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MAJOR [Masthead.class.php#L3](https://github.com/Wikia/app/blob/2eaed17829db7877bb37d81d0dd9398498b75871/extensions/wikia/UserProfilePageV3/Masthead.class.php#L3): Class "Masthead" has 30 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
1. MAJOR [Masthead.class.php#L25](https://github.com/Wikia/app/blob/2eaed17829db7877bb37d81d0dd9398498b75871/extensions/wikia/UserProfilePageV3/Masthead.class.php#L25): Remove this unused "$mMsgCityId" private field. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1068)
1. MAJOR [Masthead.class.php#L172](https://github.com/Wikia/app/blob/2eaed17829db7877bb37d81d0dd9398498b75871/extensions/wikia/UserProfilePageV3/Masthead.class.php#L172): Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. MAJOR [Masthead.class.php#L174](https://github.com/Wikia/app/blob/2eaed17829db7877bb37d81d0dd9398498b75871/extensions/wikia/UserProfilePageV3/Masthead.class.php#L174): Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
1. MAJOR [Masthead.class.php#L382](https://github.com/Wikia/app/blob/2eaed17829db7877bb37d81d0dd9398498b75871/extensions/wikia/UserProfilePageV3/Masthead.class.php#L382): Remove the unused function parameter "$tracker". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
1. MAJOR [Masthead.class.php#L438](https://github.com/Wikia/app/blob/2eaed17829db7877bb37d81d0dd9398498b75871/extensions/wikia/UserProfilePageV3/Masthead.class.php#L438): Replace "and" with "&&". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2010)
1. MAJOR [Masthead.class.php#L521](https://github.com/Wikia/app/blob/2eaed17829db7877bb37d81d0dd9398498b75871/extensions/wikia/UserProfilePageV3/Masthead.class.php#L521): Remove this unused private "getThumbPath" method. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1144)
1. MAJOR [Masthead.class.php#L594](https://github.com/Wikia/app/blob/2eaed17829db7877bb37d81d0dd9398498b75871/extensions/wikia/UserProfilePageV3/Masthead.class.php#L594): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
1. MAJOR [Masthead.class.php#L663](https://github.com/Wikia/app/blob/2eaed17829db7877bb37d81d0dd9398498b75871/extensions/wikia/UserProfilePageV3/Masthead.class.php#L663): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
1. MAJOR [Masthead.class.php#L672](https://github.com/Wikia/app/blob/2eaed17829db7877bb37d81d0dd9398498b75871/extensions/wikia/UserProfilePageV3/Masthead.class.php#L672): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 19 issues
- MAJOR 12 major
- MINOR 1 minor
- INFO 6 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:
1. MAJOR [Masthead.class.php#L3](https://github.com/Wikia/app/blob/e00ed5011e1dc0262bd276705043175ed5c108b6/extensions/wikia/UserProfilePageV3/Masthead.class.php#L3): Class "Masthead" has 29 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1448)
2. MAJOR [Masthead.class.php#L25](https://github.com/Wikia/app/blob/e00ed5011e1dc0262bd276705043175ed5c108b6/extensions/wikia/UserProfilePageV3/Masthead.class.php#L25): Remove this unused "$mMsgCityId" private field. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1068)
3. MAJOR [Masthead.class.php#L172](https://github.com/Wikia/app/blob/e00ed5011e1dc0262bd276705043175ed5c108b6/extensions/wikia/UserProfilePageV3/Masthead.class.php#L172): Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
4. MAJOR [Masthead.class.php#L174](https://github.com/Wikia/app/blob/e00ed5011e1dc0262bd276705043175ed5c108b6/extensions/wikia/UserProfilePageV3/Masthead.class.php#L174): Add curly braces around the nested statement(s). [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS121)
5. MAJOR [Masthead.class.php#L382](https://github.com/Wikia/app/blob/e00ed5011e1dc0262bd276705043175ed5c108b6/extensions/wikia/UserProfilePageV3/Masthead.class.php#L382): Remove the unused function parameter "$tracker". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)
6. MAJOR [Masthead.class.php#L438](https://github.com/Wikia/app/blob/e00ed5011e1dc0262bd276705043175ed5c108b6/extensions/wikia/UserProfilePageV3/Masthead.class.php#L438): Replace "and" with "&&". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2010)
7. MAJOR [Masthead.class.php#L590](https://github.com/Wikia/app/blob/e00ed5011e1dc0262bd276705043175ed5c108b6/extensions/wikia/UserProfilePageV3/Masthead.class.php#L590): Reduce the number of returns of this function 4, down to the maximum allowed 3. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1142)
8. MAJOR [Masthead.class.php#L659](https://github.com/Wikia/app/blob/e00ed5011e1dc0262bd276705043175ed5c108b6/extensions/wikia/UserProfilePageV3/Masthead.class.php#L659): Remove this commented out code. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS125)
9. MAJOR [Masthead.class.php#L668](https://github.com/Wikia/app/blob/e00ed5011e1dc0262bd276705043175ed5c108b6/extensions/wikia/UserProfilePageV3/Masthead.class.php#L668): Add a "case default" clause to this "switch" statement. [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS131)
10. MAJOR [Masthead.class.php#L741](https://github.com/Wikia/app/blob/e00ed5011e1dc0262bd276705043175ed5c108b6/extensions/wikia/UserProfilePageV3/Masthead.class.php#L741): Remove the unused function parameters "$article, $text, $summary, $minoredit, $watchthis, $sectionanchor, $flags, $revision, $baseRevId". [rule](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1172)

**Pull request merged**